### ISSUE TITLE
search: no mapping found for [_created] in order to sort on

### ISSUE THREAD

**Issue by dosydon** [zenodo/zenodo#1027]
# Problem
I'm having a problem installing zenodo for development. I've followed this instruction (http://zenodo.readthedocs.io/en/latest/installation.html#development-installation). I also checked out the troubleshooting page (https://github.com/zenodo/zenodo/wiki/Troubleshooting), but my issue doesn't seem to be addressed.

My installation worked fine initially, but I started getting the following errors with elastic search next time I ran zenodo locally.  zenodo renders everything fine except for records as it seems to fail to retrieve them. 
I tried recreating docker containers and virtualenv without luck.

```
parse search source [{"sort": [{"_created": {"order": "desc"}}], "query": {"bool": {"filter": [{"query_string": {"query": "communities:zenodo AND access_right:open"}}]}}, "from": 0, "size": 10}]]; nested: SearchParseException[No mapping found for [_created] in order to sort on];
es_1    | Caused by: SearchParseException[failed to parse search source [{"sort": [{"_created": {"order": "desc"}}], "query": {"bool": {"filter": [{"query_string": {"query": "communities:zenodo AND access_right:open"}}]}}, "from": 0, "size": 10}]]; nested: SearchParseException[No mapping found for [_created] in order to sort on];
es_1    |       at org.elasticsearch.search.SearchService.parseSource(SearchService.java:855)
es_1    |       at org.elasticsearch.search.SearchService.createContext(SearchService.java:654)
es_1    |       at org.elasticsearch.search.SearchService.createAndPutContext(SearchService.java:620)
es_1    |       at org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:371)
es_1    |       at org.elasticsearch.search.action.SearchServiceTransportAction$SearchQueryTransportHandler.messageReceived(SearchServiceTransportAction.java:368)
es_1    |       at org.elasticsearch.search.action.SearchServiceTransportAction$SearchQueryTransportHandler.messageReceived(SearchServiceTransportAction.java:365)
es_1    |       at org.elasticsearch.transport.TransportRequestHandler.messageReceived(TransportRequestHandler.java:33)
es_1    |       at org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:75)
es_1    |       at org.elasticsearch.transport.TransportService$4.doRun(TransportService.java:376)
es_1    |       at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
es_1    |       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
es_1    |       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
es_1    |       at java.lang.Thread.run(Thread.java:745)
es_1    | Caused by: SearchParseException[No mapping found for [_created] in order to sort on]
es_1    |       at org.elasticsearch.search.sort.SortParseElement.addSortField(SortParseElement.java:213)
es_1    |       at org.elasticsearch.search.sort.SortParseElement.addCompoundSortField(SortParseElement.java:187)
es_1    |       at org.elasticsearch.search.sort.SortParseElement.parse(SortParseElement.java:85)
es_1    |       at org.elasticsearch.search.SearchService.parseSource(SearchService.java:838)
es_1    |       ... 12 more
```

# My Environments
- OS (Ubuntu 16.04LTS)
- I used docker for DB, ES, MQ, and Cache.
- Docker version 17.03.0-ce, build 60ccb22
- I'm working on master branch ( 5f00017757d3f2fa3e942f9c49cc636f66e7a35a)

**Comment by krzysztof** [comment]
> My installation worked fine initially, but I started getting the following errors with elastic search next time I ran zenodo locally. zenodo renders everything fine except for records as it seems to fail to retrieve them.

Is this showing on the frontpage only or also on search results?

> I tried recreating docker containers and virtualenv without luck.

This looks like an issue with ES index/mapping, what usually works is to recreate the index:
``zenodo index destroy --force``
``zenodo index init``
``zenodo migration reindex -t recid``
``zenodo migration reindex -t od_lic``
``zenodo index run``

Recreating docker containers would actually solve that as well, but I guess by recreate you mean you restarted docker containers? (If so, the state of ES, DB and everything else stays the same, so the problem will persist).
If at any point you want to truly recreate the docker services you need to *remove* them and the rebuilt:
``docker-compose rm es mq cache db`` (destroys the containers)
``docker-compose up es mq cache db`` ("up" will first build them again here. Note that DB and ES are empty at this point, so you need to create initialize everything from scratch again).

**Comment by dosydon** [comment]
@krzysztof  Thanks for your response, but I'm still getting the same error message after recreating the index (the results are now displayed ok somehow, though). 

As I look into this problem in more detail, I noticed that the current head of the master branch (https://github.com/zenodo/zenodo/commit/5f00017757d3f2fa3e942f9c49cc636f66e7a35a) also gave the same error when running on Travis (https://travis-ci.org/zenodo/zenodo/jobs/214250654) and seemed to miss some test coverages due to the error (https://coveralls.io/builds/10737563). 

Is this a known issue I can safely ignore? Sorry if I'm missing previous discussions on this matter.

Some More Details:
> Is this showing on the frontpage only or also on search results?

It seems like I'm getting the error message only on the front page but not on search results. 

> Recreating docker containers would actually solve that as well, but I guess by recreate you mean you > just restarted docker containers? If so, the state of ES, DB and everything else stays the same, so the > problem will persist.

I did recreate all the docker containers rather than just stopping and then restarting them.
